### PR TITLE
[risk=no][no ticket] Updates to system requirements readme

### PR DESCRIPTION
**Pull request by jmthibault79**
A few tweaks I noticed as I was setting up a new computer

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/docs/developer-system-requirements.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ We currently only support MacOS and Linux for development and testing.
   * [Ruby](https://www.ruby-lang.org/en/downloads/)
     * Our team's dev/ops scripts are written in Ruby. Most common operations are launched via the project.rb script at the root of each sub-project.
     * You may need to add ruby gems to your path if things are not working. If so follow [these instructions](https://guides.rubygems.org/faqs/#i-installed-gems-with---user-install-and-their-commands-are-not-available).
-  * [Python](https://www.python.org/downloads/) >= 2.7.9
+  * [Python](https://www.python.org/downloads/) >= 3.5
```
Comment (jmthibault79) [inline api/docs/developer-system-requirements.md:17]: "Supported versions are Python 3 (3.5 to 3.9)"
https://cloud.google.com/sdk/docs/install

File: api/docs/developer-system-requirements.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 ### Operating System
 We currently only support MacOS and Linux for development and testing.
 ### Prerequisite Software
-  * [Java 8](https://openjdk.java.net/projects/jdk8/)
+  * [Java 8](https://adoptium.net/temurin/releases/)
```
Comment (jmthibault79) [inline api/docs/developer-system-requirements.md:5]: "Our July 2021 and future releases will come from [Adoptium.net](https://www.adoptium.net/)"
https://adoptopenjdk.net/

File: api/docs/developer-system-requirements.md [hunk 3]
```diff
@@ -26,13 +26,10 @@ We currently only support MacOS and Linux for development and testing.
     gcloud components install alpha
     ```
 
-
 For local development, also install:
 
   * [yarn](https://yarnpkg.com/lang/en/docs/install/#mac-stable)
-  * [Node.js](https://nodejs.org/en/) >= 14.  Currently known to work up to 16.15.1
-  * [Java 8](https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
```
Comment (jmthibault79) [inline api/docs/developer-system-requirements.md:34]: redundant w/ above

File: api/docs/developer-system-requirements.md [hunk 4]
```diff
@@ -26,13 +26,10 @@ We currently only support MacOS and Linux for development and testing.
...
-  * [Docker Compose](https://docs.docker.com/compose/install/)
+  * [Node.js](https://nodejs.org/en/) >= 14.  Currently known to work up to 18.12.1
```
Comment (jmthibault79) [inline api/docs/developer-system-requirements.md:32]: I installed Node 18.12.1 and it worked.

**Review by jmthibault79 — COMMENTED** [review]
I'm up and running on my new computer, so these instructions still work :)

File: api/docs/developer-system-requirements.md (hunk 1 above)
Comment (Peter-Lavigne) [reply api/docs/developer-system-requirements.md:17]: We have a few other python files but they all appear to be python 3 👍

**Review by Peter-Lavigne — APPROVED** [review]

**Pull request merged**